HighwayPatrol.cpp

```diff
@@ -28,10 +35,32 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
     std::cout << name << ": vehicle is traveling " << v->speed - h->speedLimit << " miles per hour over the speed limit" << std::endl;
     if( willArrest )
     {
-        assert(false);
+        std::string vehicleType = findVehicleType(v); // find type of vehicle, function below
+        
         //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+        std::cout << name << ": YOU IN THE [ " << vehicleType << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
         std::cout << "EVERYONE ELSE, SLOW DOWN!! \n\n\n";
         h->removeVehicle(v);
     }
 }
+
+std::string HighwayPatrol::findVehicleType(Vehicle* v) // once again dynamic cast to find vehicleType
+{
+    if( auto* car = dynamic_cast<Car*>(v))
+    {
+        return("car");
+    }
+    else if ( auto* motorcycle = dynamic_cast<Motorcycle*>(v))
+    {
+        return("motorcycle");
+    }
+    else if ( auto* truck = dynamic_cast<SemiTruck*>(v))
+    {
+        return("truck");
+    }
+    else
+    {
+        return("unknown type");
+    }    
```

Comment:
this `else` is not needed.

If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise